File: backend/internal/models/route.go
package models

import "time"

const (
	RouteStatusPlanned   = "planned"
	RouteStatusActive    = "active"
	RouteStatusCompleted = "completed"
)

type Route struct {
	ID               string    `json:"id"`
	TruckID          string    `json:"truck_id"`
	OrderedPointIDs  []string  `json:"ordered_point_ids"`
	DistanceKm       float64   `json:"distance_km"`
	EstimatedFuelL   float64   `json:"estimated_fuel_l"`
	EstimatedMinutes int       `json:"estimated_minutes"`
	Status           string    `json:"status"`
	CreatedAt        time.Time `json:"created_at"`
}

type RouteStop struct {
	Order      int        `json:"order"`
	WastePoint WastePoint `json:"waste_point"`
}

type OptimizationResult struct {
	RouteID          string      `json:"route_id"`
	TruckID          string      `json:"truck_id"`
	Stops            []RouteStop `json:"stops"`
	DistanceKm       float64     `json:"distance_km"`
	EstimatedMinutes int         `json:"estimated_minutes"`
	EstimatedFuelL   float64     `json:"estimated_fuel_l"`

	BaselineDistanceKm float64 `json:"baseline_distance_km"`
	BaselineMinutes    int     `json:"baseline_minutes"`
	BaselineFuelL      float64 `json:"baseline_fuel_l"`
	DistanceSavedKm    float64 `json:"distance_saved_km"`
	FuelSavedL         float64 `json:"fuel_saved_l"`
	TimeSavedMinutes   int     `json:"time_saved_minutes"`
}

File: frontend/src/types/route.js
/**
 * @typedef {'planned'|'active'|'completed'} RouteStatus
 *
 * @typedef {Object} Route
 * @property {string} id
 * @property {string} truck_id
 * @property {string[]} ordered_point_ids
 * @property {number} distance_km
 * @property {number} baseline_distance_km
 * @property {number} estimated_fuel_l
 * @property {number} estimated_minutes
 * @property {RouteStatus} status
 * @property {string} created_at
 *
 * @typedef {Object} RouteStop
 * @property {number} order
 * @property {import('./wastePoint').WastePoint} waste_point
 *
 * @typedef {Object} OptimizationResult
 * @property {string} route_id
 * @property {string} truck_id
 * @property {RouteStop[]} stops
 * @property {number} distance_km
 * @property {number} estimated_minutes
 * @property {number} estimated_fuel_l
 * @property {number} baseline_distance_km
 * @property {number} baseline_minutes
 * @property {number} baseline_fuel_l
 * @property {number} distance_saved_km
 * @property {number} fuel_saved_l
 * @property {number} time_saved_minutes
 */

export {}
